### PR TITLE
Update usage of PyTorch's custom op API

### PR DESCRIPTION
**Pull request by zou3519**
# Description

Hi, I maintain the custom ops story in PyTorch. This PR updates the the usage of PyTorch's private custom op API to a newer API. This API is still private but closer to what we want it to be.

## Type of change

API migration.

### CODE REVIEW

**Review by gs-olive — COMMENTED** [review]
Changes look good - thanks for this update. Added a few comments/questions. Additionally, please run the Python linting on this PR. You can do so by running the following from the root directory of the repository:
```bash
pip install -r requirements-dev.txt
pre-commit install
```
Subsequent commits will be automatically linted.

This PR is currently blocked pending a Torch nightly upgrade on the repository.

**Comment by gs-olive** [comment]
Hi @zou3519 - we recently updated the stack to use the latest nightly distribution of PyTorch. Could you run linting and rebase to the latest main? Thank you

**Comment by zou3519** [comment]
Will do!

**Comment by gs-olive** [comment]
Fixes #2328

File: py/torch_tensorrt/dynamo/lowering/substitutions/einsum.py [hunk 1]
```diff
@@ -1,26 +1,21 @@
...
 from torch_tensorrt.dynamo.lowering._pre_aot_lowering import register_substitution
 from torch_tensorrt.fx.converter_registry import tensorrt_converter
 from torch_tensorrt.fx.converters.converter_utils import set_layer_name
 from torch_tensorrt.fx.types import TRTNetwork, TRTTensor
 
 
-@custom_op(
-    qualname="tensorrt::einsum",
-    manual_schema="(str equation, Tensor[] tensors) -> Tensor",
+library.custom_op(
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/lowering/substitutions/einsum.py:12]: Remove one space above this call; add one more space below line 15

File: py/torch_tensorrt/dynamo/lowering/substitutions/maxpool1d.py [hunk 2]
```diff
@@ -20,14 +20,10 @@
 # types. The namespace, such as tensorrt, will cause the op to be registered as torch.ops.tensorrt.your_op
 # Then, create a placeholder function with no operations, but having the same schema and naming as that
 # used in the decorator
-@custom_op(
-    qualname="tensorrt::maxpool1d",
-    manual_schema="(Tensor x, int[1] kernel_size, int[1] stride, int[1] padding, int[1] dilation, bool ceil_mode) -> Tensor",
+library.custom_op(
+    "tensorrt::maxpool1d",
+    "(Tensor x, int[1] kernel_size, int[1] stride, int[1] padding, int[1] dilation, bool ceil_mode) -> Tensor"
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/lowering/substitutions/maxpool1d.py:25]: Add comma at end of this line; add another space below line 26

**Review by gs-olive — COMMENTED** [review]
Looks good! Just pending the Python linting fixes mentioned in the comments below.

**Comment by zou3519** [comment]
Thanks; I'm not actually sure how to run the linter (do I need to undo the commit and then redo it? for the pre-commit hook to pick it up?)

**Comment by gs-olive** [comment]
Yes, you would need to somehow change those files for the pre-commit hook to track it. Any small change to the file (like an additional newline at the end) will cause the whole file to lint, which also works.

**Comment by zou3519** [comment]
> Yes, you would need to somehow change those files for the pre-commit hook to track it. Any small change to the file (like an additional newline at the end) will cause the whole file to lint, which also works.

Perfect, that did the trick (newline at the end)

**Comment by gs-olive** [comment]
Great! I think `py/torch_tensorrt/dynamo/lowering/substitutions/maxpool1d.py` might need it as well, then will merge.

**Comment by gs-olive** [comment]
@zou3519 - I reran the tests and will merge once they pass - I don't believe the failures are related to the content of this PR

**Review by gs-olive — APPROVED** [review]
Looks good to me! Now passing CI

**Pull request merged**